src/t8_geometry/t8_geometry_implementations/t8_geometry_occ.cxx

```diff
@@ -0,0 +1,242 @@
+  out_coords[1] += edge_delta[1];
+  out_coords[2] += edge_delta[2];
+  
+  /* Check each face for geometry */
+  for (int i = 0; i < 6; ++i)
+  {
+    if (faces[i] >= 0)
+    {
+      /* Interpolate coordinates without between face vertices */
+      for (int j = 0; j < 3; ++j)
+      {
+        interpolated_coords[j] = ((1 - ref_coords[normal_sequence[int(i / 2.0)][0]]) * active_tree_vertices[t8_face_vertex_to_tree_vertex[T8_ECLASS_HEX][i][0] * 3 + j]
+                                + ref_coords[normal_sequence[int(i / 2.0)][0]] * active_tree_vertices[t8_face_vertex_to_tree_vertex[T8_ECLASS_HEX][i][1] * 3 + j])
+                                * (1 - ref_coords[normal_sequence[int(i / 2.0)][1]]);
+        interpolated_coords[j] += ((1 - ref_coords[normal_sequence[int(i / 2.0)][0]]) * active_tree_vertices[t8_face_vertex_to_tree_vertex[T8_ECLASS_HEX][i][2] * 3 + j]
+                                + ref_coords[normal_sequence[int(i / 2.0)][0]] * active_tree_vertices[t8_face_vertex_to_tree_vertex[T8_ECLASS_HEX][i][3] * 3 + j])
+                                * ref_coords[normal_sequence[int(i / 2.0)][1]];
+      }
+      
+      /* Interpolate parameters between face vertices */
+      const double *parameters = (double *) t8_cmesh_get_attribute(cmesh, t8_get_package_id (),
+                                                      T8_CMESH_OCC_SURFACE_PARAMETERS_ATTRIBUTE_KEY + i,
+                                                      gtreeid);
+      for (int j = 0; j < 2; ++j)
+      {
+        param[j] = (parameters[0 + j] * (1 - ref_coords[normal_sequence[(int)(i / 2.0)][0]])
+                  + parameters[2 + j] * (ref_coords[normal_sequence[(int)(i / 2.0)][0]]))
+                  * (1 - ref_coords[normal_sequence[(int)(i / 2.0)][1]]);
+        param[j] += (parameters[4 + j] * (1 - ref_coords[normal_sequence[(int)(i / 2.0)][0]])
+                    +parameters[6 + j] * (ref_coords[normal_sequence[(int)(i / 2.0)][0]]))
+                    *(ref_coords[normal_sequence[(int)(i / 2.0)][1]]);
+      }
+      
+      /* Check if calculated parameters are valid */
+      T8_ASSERT(!t8_global_occ_surface[faces[i]].IsNull());
+      double u1, u2, v1, v2;
+      t8_global_occ_surface[faces[i]]->Bounds(u1, u2, v1, v2);
+      T8_ASSERT(((u1 <= param[0] && param[0] <= u2) || (u2 <= param[0] && param[0] <= u1))
+                && ((v1 <= param[1] && param[1] <= v2) || (v2 <= param[1] && param[1] <= v1)));
+
+      /* Compute displacement between vertex interpolation and surface evaluation with interpolated parameters */
+      t8_global_occ_surface[faces[i]]->D0(param[0], param[1], pnt);
+      surface_deltas[i][0] = pnt.X() - interpolated_coords[0];
+      surface_deltas[i][1] = pnt.Y() - interpolated_coords[1];
+      surface_deltas[i][2] = pnt.Z() - interpolated_coords[2]; 
+    }
+  }  
+
+  /* Multiply surface displacements with corresponding ref coords and add them to the out coords */
+  for (int i = 0; i < 3; ++i)
```

Comment:
why 3?
Also rename i to something readable.

Reply:
see 2fe3e66